supabase/client.py

```
@@ -0,0 +1,36 @@
+    """Return a singleton synchronous-ish Supabase client.
+
+    The supabase-py client exposes both sync and async interfaces depending on
+    usage. For simple migration scripts and lightweight usage we return the
+    regular client. If you need full async support, adapt calls to
+    client.postgrest.rpc/ etc. or use an async HTTP transport as documented by
+    supabase-py.
+    """
+    global _client
+    if _client is None:
+        if not SUPABASE_URL or not SUPABASE_KEY:
+            raise RuntimeError("SUPABASE_URL and SUPABASE_KEY must be set in environment")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)